File: SpecialRel.py
#!/usr/bin/env python
import sys, os, math

c = 299792458

def main():
    print("SpecialRel.py has been loaded.")

def ctometer(a):    # Converts the speed that is in terms of c to meters per second.
    return a * c

def metertoc(a):    # Converts the speeed that is in terms of meters per second into terms of c.
    return float(a / c)

def gamma(v):    # Calculates the gamma taking the speed in terms of c.
    return float((1 / math.sqrt((1 - (v**2)))))

def ctogamma(v):  # Calculates the value of gamma based on the speed in terms of meters per second.
    return float(gamma(metertoc(v)))

def xlorentz(speed, x, t):  # Calculates the lorentz transformation for space.
    return float(gamma(speed) * (x - speed * t))

def tlorentz(speed, x, t):  # Calculates the lorentz transformation for time.
    return float(gamma(speed) * (t - speed * x))

def vlorentz(v, w):  # Calculates the lorentz transformation for velocity.
    return float((v - w) / (1 - (v * w)))

def timedialation(v, t):  # Calculates the time dialation based on the velocity and time.
    return float(gamma(v) * t)

def lengthcontraction(v, h):   # Calculates the length contraction based on the velocity and time.
    return float(h / (gamma(v)))

def spacetimeinterval(x, t):   # Calculates the spacetime interval based on the space and time.
    return float(math.sqrt((t**2) -(x**2))))

def spacetimeinterval2(t1, t2, x1, x2):
    return float(math.sqrt((t2 - t1)**2 - (x2 - x1)**2))

def spaceinterval(x1, x2):
    return float(x2 - x1)

def timeinterval(t1, t2):
    return float(t2 - t1)

if __name__=="__main__":
    main()

File: SpecialRel.clj
(ns specialrel.core)

(def c 299792458) ;; Speed of light constant

(defn ctometer [x] ;; converts to meters per second
  (* x c))

(defn metertoc [x] ;; converts meters per second to terms of c
  (float (/ x c)))

(defn gamma [v]   ;; calculates the gamma factor depending on the velocity in terms of c
  (/ 1 (float (Math/sqrt (- 1 (* v v))))))

(defn ctogamma [v]  ;; converts to gamma depending on the velocity in terms of meters per second
  (gamma (float (metertoc v))))

(defn xlorentz [speed x t]  ;; Does a lorentz transformation for space
  (* (gamma speed) (- x (* speed t))))

(defn tlorentz [speed x t]  ;; Does a lorentz transformation for time
  (* (gamma speed) (- t (* speed x))))

(defn vlorentz [v w]  ;; Lorentz transformation for velocity
  (/ (- v w) (- 1 (* v w))))

(defn lorentz [speed x t]  ;; Lorentz transformation for both space and time
  (let [x (xlorentz speed x t)
        t (tlorentz speed x t)]
    (println "x'=" x "  t'=" t)
    [x, t]))

(defn timedialation [v t]  ;; Calculates the time dialation given the velocity and time
  (* t (gamma v)))

(defn lengthcontraction [v h] ;; Calculates the length contraction based on velocity and length
  (/ h (gamma v)))

(defn spacetimeinterval [x t] ;; calculates the spacetime interval given the space and time in meters.
  (Math/sqrt (- (* t t) (* x x))))

(defn spaceinterval [x1 x2]
  (- x2 x1))

(defn timeinterval [t1 t2]
  (t2 - t1))

(defn spacetimeinterval2 [t1 t2 x1 x2]
  (Math/sqrt (- (Math/pow (- t2 t1) 2) (Math/pow (- x2 x1) 2))))

(def planck-constant 6.626e-34)

(def mass-of-proton 1.67e-27)

(def mass-of-electron 9.1e-31)

(def elementary-charge 1.602e-19)

(def electron-volt-per-c-square-in-kg 1.783e-36)

(def amu 1.66e-27)

(defn energy [wavelength]
  (/ (* planck-constant c)
     wavelength))

(defn rel-momentum [p m]
  (Math/sqrt (/ (Math/pow (/ p m) 2)
                (+ 1 (/ (Math/pow (/ p m) 2) 1)))))
